src/configuration.cc

```diff
@@ -132,10 +133,15 @@ int MetadataAgentConfiguration::ParseArguments(int ac, char** av) {
 }
 
 void MetadataAgentConfiguration::ParseConfigFile(const std::string& filename) {
-  std::lock_guard<std::mutex> lock(mutex_);
   if (filename.empty()) return;
 
-  YAML::Node config = YAML::LoadFile(filename);
+  std::ifstream ifs(filename);
```

Comment:
Let's replace `ifs` with `input` -- it reads better, IMO.

Reply:
Agreed, done.